src/Config/RuntimeConfigLoader.cs

```diff
@@ -31,6 +31,12 @@ public abstract class RuntimeConfigLoader
     // state in place of using out params.
     public RuntimeConfig? RuntimeConfig;
 
+    public RuntimeConfig? lastValidRuntimeConfig;
```

Comment:
All these properties are public properties. So, naming convention wise they should be capitalized. LastValidRuntimeConfig, IsNewConfigDetected, IsNewConfigValidated.